sdk/lib/atl/atlimage.h

```diff
@@ -1213,7 +1226,34 @@ class CImage
 
 DECLSPEC_SELECTANY CImage::CInitGDIPlus CImage::s_gdiplus;
 
-}
+class CImageDC
+{
+private:
+    const CImage& m_image;
+    HDC m_hDC;
+
+public:
+    CImageDC(const CImage& image)
+        : m_image(image)
+        , m_hDC(image.GetDC())
+    {
+    }
+
+    virtual ~CImageDC() throw()
+    {
+        m_image.ReleaseDC();
+    }
+
+    operator HDC() const throw()
+    {
+        return m_hDC;
+    }
+
+    CImageDC(const CImageDC&) = delete;
```

Comment:
Even if they are `= delete`'d, isn't it possible to put them in a `private:` section?
(OK I see you removed the private due to Serge's comment, but I'm not sure his comment was about actually removing the private keyword)

Reply:
> Even if they are = delete'd, isn't it possible to put them in a private: section?

Yes, it's possible.

Reply:
> not sure

Yes, I did mean using `public:` is more explicit than `private:`.
(And I suggested to move these declarations too.)

Reply:
>  I did mean using public: is more explicit than private:.

How can this be "more explicit" than private? The aim is to trigger a compile error whenever those constructors are being used. Using private would do the job, already. (Besides the C++11 way with `= delete;` that really explicitly say that no implementation should be emitted https://en.cppreference.com/w/cpp/language/function#Deleted_functions .)

Reply:
Improved documentation!?
https://learn.microsoft.com/en-us/cpp/cpp/explicitly-defaulted-and-deleted-functions?view=msvc-170
'The copy constructor and copy-assignment operator are public but deleted. It is a compile-time error to define or call a deleted function.'
I had found another page about that, but can't find it again...

Reply:
>  'The copy constructor and copy-assignment operator are public but deleted. It is a compile-time error to define or call a deleted function.'

The MSDN article was just describing what was shown in the example above that text. Their aim (with that example) was to underline that it was not _mandatory_ anymore to put the cast-constructor and the operator= inside a `private:` section because the `=delete` syntax already does the wanted job.
But this doesn't mean that putting them into a `private:` (in addition to having the `=delete` part) wouldn't work...

Reply:
> not mandatory anymore

Exactly my point, as in not needed / not wanted / should not / counterproductive...

> wouldn't work

I never wrote `private:` wouldn't work.